File: Day01/Part1.ts
const text = await Deno.readTextFile("./Day01/input.txt")

const answer = text
    .split("\r\n")
    .map(item => [...item].filter(char => !isNaN(Number(char))))
    .filter(numbers => numbers.length >= 1)
    .map(numbers => Number(`${numbers[0]}${numbers[numbers.length - 1]}`))
    .reduce((acc, state) => acc += state, 0)

console.log(answer)

File: Day01/Part1.cs
using System;
using System.IO;

var text = await File.ReadAllTextAsync(@"Day01/input.txt");

var answer = text
    .Split("\r\n")
    .Select(item => item.Where(c => int.TryParse($"{c}", out _)).Select(c => int.Parse($"{c}")))
    .Where(item => item.Count() > 0)
    .Aggregate(0, (acc, state) => acc + state.First() * 10 + state.Last());

Console.WriteLine(answer);